prog/vm/github_runner.rb

```diff
@@ -189,14 +190,24 @@ def before_run
   end
 
   label def wait
+    # STATE_MAP is defined in rhizome/common/bin/daemonizer script.
+    #   dead    => NotStarted
+    #   running => InProgress
+    #   exited  => Succeeded
+    #   failed  => Failed
     case vm.sshable.cmd("systemctl show -p SubState --value #{SERVICE_NAME}").chomp
     when "exited"
       github_runner.incr_destroy
       nap 0
     when "failed"
+      Clog.emit("#{SERVICE_NAME} is failed, registering the runner again") { {github_runner: github_runner.values} }
+      vm.sshable.cmd("sudo systemctl stop runner-script")
+      vm.sshable.cmd("./actions-runner/config.sh remove --local")
       github_client.delete("/repos/#{github_runner.repository_name}/actions/runners/#{github_runner.runner_id}")
       github_runner.update(runner_id: nil, ready_at: nil)
       hop_register_runner
+    when "dead"
```

Comment:
is it possible for this to exhaust all of the VMs in the pool? Let's assume there is a bug and the script is constantly failing, would that cause a continuous VM creation/deletion?

Reply:
I removed this case